src/arrayMethodReduce.js

```diff
@@ -5,8 +5,16 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = arguments.length !== 1
+      ? callback(initialValue, this[0], 0, this) : this[0];
```

Comment:
Try to call `[].reduce2((sum, num) => sum + num, 0);` with your reduce2 and reduce methods. The result will be different )

Reply:
Ok, i don`t see the difference. But anyway i rewrote this to more readable logic

Reply:
on your logic result will be NaN, but should be 0 )

Reply:
Oh, sorry. I didn`t test it with empty array. My bad